apis/v1alpha2/validation/gateway_test.go

```diff
@@ -77,6 +77,24 @@ func TestValidateGateway(t *testing.T) {
 			},
 			expectErrsOnFields: []string{},
 		},
+		"tls config present with tls protocol": {
+			mutate: func(gw *gatewayv1a2.Gateway) {
+				protocol := gatewayv1a2.TLSProtocolType
+				tlsConfig := gatewayv1a2.GatewayTLSConfig{}
+				gw.Spec.Listeners[0].Protocol = protocol
+				gw.Spec.Listeners[0].TLS = &tlsConfig
```

Comment:
Would this work? If so, I'd recommend updating all the similar tests below as well.
```suggestion
				gw.Spec.Listeners[0].Protocol = gatewayv1a2.TLSProtocolType
				gw.Spec.Listeners[0].TLS = &gatewayv1a2.GatewayTLSConfig{}
```

Reply:
Declared tlsConfig as a global variable for all tests so that we can use that directly over there as we are not modifying that